core/base/inc/TVirtualMutex.h

```diff
@@ -94,7 +94,7 @@ class TLockGuard {
 // TMutex m; R__LOCKGUARD(&m);
 // Warning: if program is compiled without pthread support, _REENTRANT will
 // be undefined and the macro has (silently) no effect, no locks are performed.
-#if defined (_REENTRANT) || defined (WIN32)
+#if defined (_REENTRANT) || defined (WIN32) || defined (R__FBSD)
```

Comment:
I think for good measure you should apply the same change to `core/base/inc/TVirtualRWMutex.h`, `core/cont/inc/TCollection.h`, and `core/meta/inc/TInterpreter.h`